### ISSUE TITLE
ClipToBounds on parent control is not clipping childrens visual while enabled Compositor

### ISSUE THREAD

**Issue by appleneko2001** [AvaloniaUI/Avalonia#8916]
**Describe the bug**
In avaloniaUI you can clip childrens by enabling `ClipToBounds` property on parent control, additional supports clipping rounded corner. But there was a problem while I want clip corner in AvaloniaUI 11.0, it completely broken and you could see some unclipped corner on parent visual. 

**To Reproduce**

1. Create a new project with Avalonia package, version `11.0-preview1`
2. Add a control, might be border, or any control supports `CornerRadius` property and clipping
3. Add some childrens in parent control (with `Panel` wrapping), and fill them with colors
4. Enable `ClipToBounds` in parent control, set `CornerRadius` 16 (or whatever numbers you want, but minimum 4)
5. Run program, you will see such glitch I described in this issue.

or you can use project from [my glitch reproduce repository](https://github.com/appleneko2001/AvaloniaReproduceClipGlitch1/tree/development-branch/Avalonia.ReproduceClippingGlitch)

**Expected behavior**
Parent control or visual should clip childrens and applying rounded corners clipping with compositor enabled.
something like this:
![screenshot fragment, uses Avalonia 0.10.18, without compositor](https://user-images.githubusercontent.com/38249047/189165066-3f05e23a-b16c-41ec-aed7-107bfa1d1c03.png)

**Screenshots**
> In Material.Avalonia Demo:

![image](https://user-images.githubusercontent.com/38249047/189161420-f9d3f30c-0799-46f7-a8cd-1d27a25240bb.png)
![image](https://user-images.githubusercontent.com/38249047/189161508-75bda60b-ad8f-4bd2-9b0e-2d5abc5a185c.png)
![image](https://user-images.githubusercontent.com/38249047/189161565-861fd035-dc40-4a11-8e6f-5a2aaa563517.png)

> In reproduce project:

![image](https://user-images.githubusercontent.com/38249047/189165428-100de381-59d2-4e26-86e5-6cdf9694d9ea.png)

**Desktop (please complete the following information):**

- OS: Windows 10 Professional 21H2
- Version: Avalonia `11.0.0-preview-1`

**Additional context**
Actually it can be fixed by disable compositor, but it supposed to work as well while compositor is enabled.
Add above code to `AppBuilder` chain and you will disable compositor.
```
...
.With(new Win32PlatformOptions
{
    UseCompositor = false
})
...
```

**Comment by rabbitism** [comment]
I encountered same issue today.
Indeterminate progressbar style is broken because of this issue. 

![image](https://user-images.githubusercontent.com/14807942/190129543-f1ed1bf2-7548-42ab-a69c-8e64ff7f8713.png)

**Comment by appleneko2001** [comment]
> I encountered same issue today. Indeterminate progressbar style is broken because of this issue.
> 
> ![image](https://user-images.githubusercontent.com/14807942/190129543-f1ed1bf2-7548-42ab-a69c-8e64ff7f8713.png)

You may disable compositor (temporarily solution for this issue) at this moment, it still not ready for production.
Latest stable releases are 0.10.18. Stay on stable releases of libraries are always a best choice for production applications, especially of framework library.

**Comment by lukaszossowski** [comment]
Is there any news on this? ClipToBounds works fine in the xaml preview, but not in the app. 

Also, how to disable compositor in a mobile Android project?

**Comment by appleneko2001** [comment]
> Also, how to disable compositor in a mobile Android project?

Hi! Use `AndroidPlatformOptions` instead of `Win32PlatformOptions`:

```
.With(new AndroidPlatformOptions
{
    UseCompositor = false
})
```

Please refer [AndroidPlatform.cs#L76](https://github.com/AvaloniaUI/Avalonia/blob/e57e0b76fceb118ed15260844b4eae025662f6a6/src/Android/Avalonia.Android/AndroidPlatform.cs#L76) to know which avaloniaUI options have for Android platform

**Comment by DmitriyYukhanov** [comment]
I do confirm this is still happening at 11.0.0-preview.3 and disabling Compositor workaround helps here as well.

**Comment by kekekeks** [comment]
This should be fixed by #9543
